runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -4977,6 +4977,9 @@ inline void generateInlinedCheckCastOrInstanceOfForClass(TR::Node* node, TR_Opaq
    // Equality test
    if (!fej9->isAbstractClass(clazz) || node->getOpCodeValue() == TR::icall/*TR_checkAssignable*/)
       {
+      // For instanceof and checkcast, LHS is obtained from an instance, which cannot be abstract or interface;
+      // however, LHS for TR_checkAssignable may be abstract or interface as it may be an arbitrary class;
+      // therefore, equality test can be safely skipped for instanceof and checkcast.
```

Comment:
@0dvictor I'm not sure I follow this code and comment yet.  

The condition says:
`if (!fej9->isAbstractClass(clazz) || node->getOpCodeValue() == TR::icall/*TR_checkAssignable*/)`
If `class` is *not* abstract or it's from a `TR_checkAssignable` node, then we need to do do the equality check but the comment says that we can skip the equality check for instanceof/checkcast.

Unless I'm missing something, the comment and the code don't feel like they are in sync.  

Also, don't instanceof/checkcast need the equality check?  Or is subsumed into the superclass test below?  How else would the check of instances exact class be caught?

Reply:
FYI @andrewcraik

Reply:
Quite right my bad - I missed that - I think the confusion is in the end of the sentence. The code I believe is right. @0dvictor can you make the appropriate correction?

Reply:
@0dvictor @andrewcraik Has a PR been opened with the updated comment yet?

Reply:
@DanHeidinga Sorry that I missed the discussion in this merged PR. I've opened #2698 to update the comments.